File: javascript/fibonacci.js
// implementation with loop

let prev1 = 0;
let prev2 = 1;
const round = 18
console.log(prev1);
console.log(prev2);
for(let i = 1; i <= round; i++){
    let newFibo = prev1 + prev2
    console.log(newFibo);
    prev1 = prev2;
    prev2 = newFibo
}

// implementation with recursion

let firstFibo = 0;
let secondFibo = 1;
let count = 2;
console.log(firstFibo);
console.log(secondFibo);

function Fibonacci(prev1, prev2) {

    if(count <= 19){
        let newFibo = prev1 + prev2
        console.log(newFibo);
        prev1 = prev2;
        prev2 = newFibo
        count += 1
        Fibonacci(prev1, prev2)
    }else {
        return
    }
}

Fibonacci(firstFibo, secondFibo);

// implement with recusion find n th of fibonacci

function FindPositionOfFibonacci(n){

    if(n <= 1){
        return n
    }else {
        return FindPositionOfFibonacci(n - 1) + FindPositionOfFibonacci(n - 2)
    }
}


console.log(FindPositionOfFibonacci(19));

File: python/fibonacci.py
# implementation with loop

prev1 = 0
prev2 = 1

print(prev1)
print(prev2)
for fibo in range(18) :
    newFibo = prev1 + prev2
    print(newFibo)
    prev1 = prev2
    prev2 = newFibo
    
# implementation with Recursion

print(0)
print(1)
count = 2

def fibonacci(prev1, prev2) :
    global count
    if count <= 19 : 
        newFibo = prev1 + prev2
        print(newFibo)
        prev1 = prev2
        prev2 = newFibo
        count += 1
        fibonacci(prev1 , prev2)
    else : 
        return
    
fibonacci(0,1)

# finding value of fibonacci position with recursion

def F(n) :
    if n <= 1 :
        return n
    else :
        return F(n-1) + F(n -2)
print(F(19))
